synfig-core/src/modules/mod_svg/svg_parser.cpp

```diff
@@ -2705,6 +2888,112 @@ get_inkscape_version(const xmlpp::Element* svgNodeElement)
 	return 0; // not inkscape ;)
 }
 
+/**
+ * A parser for CSS style 'font-size'.
+ * Instantiate with parent font-size (or let the default value) and call parse() with the font-size value string.
+ */
+struct FontSize
+{
+	static constexpr float default_size = 16.; // 16.0 px
+	static constexpr float larger_factor = 1.2;
+
+	static const std::map<String, float> absolute_size_names;
+
+	float font_size = default_size;
+
+	FontSize() = default;
+	explicit FontSize(float size) : font_size(size) { }
+
+	bool parse(const String& value) {
+		if (value.empty())
+			return true;
+
+		auto length = value.length();
+
+		if (value.back() == '%') {
+			const auto factor = atof(value.substr(0, length-1).c_str()) / 100.;
+			if (approximate_less(factor, 0.))
+				return false;
+			font_size *= factor;
+			return true;
+		}
+
+		if (length >= 2) {
+
+			auto size_name_it = absolute_size_names.find(value);
+			if (size_name_it != absolute_size_names.end()) {
+				font_size = size_name_it->second;
+				return true;
+			}
+
+			if (value == "larger") {
+				larger();
+				return true;
+			}
+
+			if (value == "smaller") {
+				smaller();
+				return true;
+			}
+
+			const String unit = value.substr(length - 2, 2);
+
+			if (unit == "em") {
+				const auto factor = atof(value.substr(0, length-2).c_str());
+				if (approximate_less(factor, 0.))
+					return false;
+				font_size *= factor;
+				return true;
+			}
+
+		}
+
+		auto new_value = getDimension(value);
+		if (approximate_less(new_value, 0.))
+			return false;
+
+		font_size = new_value;
+		return true;
+	}
+
+	void larger()
+	{
+		for (auto it = absolute_size_names.begin(); it != absolute_size_names.end(); ++it) {
+			if (font_size < it->second) {
+				font_size = it->second;
+				return;
+			}
+		}
+		font_size *= larger_factor;
+	}
+
+	void smaller()
+	{
+		for (auto it = absolute_size_names.rbegin(); it != absolute_size_names.rend(); ++it) {
+			if (font_size > it->second) {
+				font_size = it->second;
+				return;
+			}
+		}
+		font_size /= larger_factor;
+	}
+
+	String to_string() const
+	{
+		return std::to_string(font_size) + "px";
+	}
+};
+
+const std::map<String, float> FontSize::absolute_size_names = {
+	{"xx-small", default_size * 3./5.},
+	{"x-small",  default_size * 3./4.},
+	{"small",    default_size * 8./9.},
+	{"medium",   default_size},
```

Comment:
Currently build fails on macOS with the error:
```
Undefined symbols for architecture x86_64:
  "FontSize::default_size", referenced from:
```

This is a little counter-intuitive, but this can be fixed by this:
```
{"medium",   default_size*1},
```

Reply:
o.O
Done.